drivers/adc/adc_nrfx_saadc.c

```diff
@@ -504,6 +573,10 @@ static void saadc_irq_handler(const struct device *dev)
 		nrf_saadc_task_trigger(NRF_SAADC, NRF_SAADC_TASK_STOP);
 		nrf_saadc_disable(NRF_SAADC);
 
+#if ADC_BUFFER_IN_RAM
+		memcpy(m_data.user_buffer, m_data.samples_buffer, m_data.ctx.sequence.buffer_size);
```

Comment:
This will not work for `options->extra_samplings` != 0. You need to store `active_channels` in `start_read()` and here do:
```suggestion
		memcpy(m_data.user_buffer, m_data.samples_buffer, m_data.active_channels);
```